File: src/TestPlainTextAssembly/ConfigFileActionSteps.cs
using System.Configuration;
using NBehave.Narrator.Framework;
using Should.Fluent;

namespace TestPlainTextAssembly
{
    [ActionSteps]
    public class ConfigFileActionSteps
    {
        private string _configValue;

        [Given(@"an assembly with a matching configuration file")]
        public void DoNothing(string action, string person)
        {
        }

        [When(@"the value of setting $key is read")]
        public void ReadConfigValue(string key)
        {
            _configValue = ConfigurationManager.AppSettings[key];
        }

        [Then("the value should be $actionPerformed")]
        public void ActionPerformed(string actionPerformed)
        {
            _configValue.Should().Equal(actionPerformed);
        }
    }
}

File: src/TestLib/ConfigFileActionSteps.fs
[<NBehave.ActionSteps>]
module TestLib.ConfigFileActionSteps
open NBehave
open System.Configuration
open Should
    
    [<Given>]
    let ``an assembly with a matching configuration file`` (action:string) (person:string) = 
        ()

    [<When>]
    let ``the value of setting $key is read`` (key:string) =
        let value = ConfigurationManager.AppSettings.[key]
        ScenarioContext.Current.Add("configValue", value)

    [<Then>]
    let ``the value should be $actionPerformed`` (actionPerformed:string) =
        let value = ScenarioContext.Current.["configValue"] |> string
        value.ShouldStartWith(actionPerformed)
